src/components/dropdown-button.tsx

```diff
@@ -121,11 +121,7 @@ const DropdownButton = ({
         style={style}
       >
         {showMenu &&
-          (childType === 'function'
-            ? (children as (
-                showMenu: Dispatch<SetStateAction<boolean>>
-              ) => ReactNode)(setShowMenu)
-            : children)}
+          (typeof children === 'function' ? children(setShowMenu) : children)}
```

Comment:
Magic!

Reply:
TypeScript magic 😅 
You were using a variable that had been set with the same statement, but TypeScript lost track of what that variable meant when you were using it here, so I just reused the statement here so that TypeScript could infer types correctly here